usaspending_api/etl/management/commands/load_transactions_in_delta.py

```diff
@@ -0,0 +1,416 @@
+                        -- in ROW_NUMBER() assigning a unique, incrementing value to each row
+                        TRUE AS assign_new_id
+                    FROM
+                        raw.detached_award_procurement
+                    WHERE
+                        updated_at >= '{last_etl_date}'
+                        AND NOT EXISTS(
+                            SELECT *
+                            FROM transaction_id_lookup
+                            WHERE transaction_id_lookup.detached_award_procurement_id = detached_award_procurement.detached_award_procurement_id
+                        )
+                )
+                UNION
+                (
+                    SELECT
+                        NULL AS detached_award_procurement_id,
+                        published_fabs_id,
+                        afa_generated_unique AS transaction_unique_id,
+                        -- A value of "true" is assigned across all rows for use in creating a partition
+                        -- in ROW_NUMBER() assigning a unique, incrementing value to each row
+                        TRUE AS assign_new_id
+                    FROM
+                        raw.published_fabs
+                    WHERE
+                        is_active IS TRUE
+                        AND updated_at >= '{last_etl_date}'
+                        AND NOT EXISTS(
+                            SELECT *
+                            FROM transaction_id_lookup
+                            WHERE transaction_id_lookup.published_fabs_id = published_fabs.published_fabs_id
+                        )
+                )
+            ) AS all_new_transactions
+        """
+        )
+
+        self.logger.info("Updating transaction_id_seq to the new max_id value")
+        max_id = self.spark.sql("SELECT MAX(id) AS max_id FROM int.transaction_id_lookup").collect()[0]["max_id"]
+        with connection.cursor() as cursor:
+            cursor.execute(f"SELECT setval('transaction_id_seq', {max_id})")
+
+    @staticmethod
+    def merge_into_sql(transaction_type):
+        if transaction_type == "fabs":
+            col_info = copy.copy(TRANSACTION_FABS_COLUMN_INFO)
+        elif transaction_type == "fpds":
+            col_info = copy.copy(TRANSACTION_FPDS_COLUMN_INFO)
+        else:
+            raise ValueError(f"Invalid value for 'transaction_type'; must select either: 'fabs' or 'fpds'")
+
+        set_cols = [f"silver_table.{col.silver_name} = source_view.{col.silver_name}" for col in col_info]
+        silver_table_cols = ", ".join([col.silver_name for col in col_info])
+
+        sql = f"""
+            MERGE INTO int.transaction_{transaction_type} AS silver_table
+            USING vw_{transaction_type}_from_source AS source_view
+            ON silver_table.transaction_id = source_view.transaction_id
+            WHEN MATCHED
+                THEN UPDATE SET
+                    {", ".join(set_cols)}
```

Comment:
Will `updated_at` (or whatever the update field is) get a new date or the "now" date when doing this?

Reply:
TODO: Need to investigate broker data and see if it is always UTC or if it could hold another timezone, as the current code would shift the time to align to UTC in the `cast_datetime_to_utc` function being used.